cmd/sourcemaps/upload.go

```diff
@@ -0,0 +1,402 @@
+	return func(cmd *cobra.Command, args []string) error {
+		u, _ := url.JoinPath(
+			viper.GetString(cliflags.BaseURIFlag),
+			"api/v2/caller-identity",
+		)
+		res, err := client.MakeRequest(
+			viper.GetString(cliflags.AccessTokenFlag),
+			"GET",
+			u,
+			"application/json",
+			nil,
+			nil,
+			false,
+		)
+		if err != nil {
+			return output.NewCmdOutputError(err, viper.GetString(cliflags.OutputFlag))
+		}
+
+		var result struct{ AccountID string }
+		if err = json.Unmarshal(res, &result); err != nil {
+			return output.NewCmdOutputError(err, viper.GetString(cliflags.OutputFlag))
+		}
+
+		projectKey := viper.GetString(cliflags.ProjectFlag)
+		u, _ = url.JoinPath(
+			viper.GetString(cliflags.BaseURIFlag),
+			"api/v2/projects",
+			projectKey,
+		)
+		res, err = client.MakeRequest(
+			viper.GetString(cliflags.AccessTokenFlag),
+			"GET",
+			u,
+			"application/json",
+			nil,
+			nil,
+			false,
+		)
+		if err != nil {
+			return output.NewCmdOutputError(err, viper.GetString(cliflags.OutputFlag))
+		}
+
+		var projectResult struct {
+			ID string `json:"_id"`
+		}
+		if err = json.Unmarshal(res, &projectResult); err != nil {
+			return output.NewCmdOutputError(err, viper.GetString(cliflags.OutputFlag))
+		}
+
+		appVersion := viper.GetString(appVersionFlag)
+		path := viper.GetString(pathFlag)
+		basePath := viper.GetString(basePathFlag)
+		backendUrl := viper.GetString(backendUrlFlag)
+
+		if backendUrl == "" {
+			backendUrl = defaultBackendUrl
+		}
+
+		highlightKey, projectID, err := verifyApiKey(result.AccountID, projectResult.ID, backendUrl)
+		if err != nil {
+			return fmt.Errorf("failed to verify API key: %w", err)
+		}
+
+		fmt.Printf("Starting to upload source maps from %s\n", path)
+
+		files, err := getAllSourceMapFiles(path)
+		if err != nil {
+			return fmt.Errorf("failed to find sourcemap files: %w", err)
+		}
+
+		if len(files) == 0 {
+			return fmt.Errorf("no source maps found in %s, is this the correct path?", path)
+		}
+
+		s3Keys := make([]string, 0, len(files))
+		for _, file := range files {
+			s3Keys = append(s3Keys, getS3Key(projectID, appVersion, basePath, file.Name))
+		}
+
+		uploadUrls, err := getSourceMapUploadUrls(highlightKey, s3Keys, backendUrl)
+		if err != nil {
+			return fmt.Errorf("failed to get upload URLs: %w", err)
+		}
+
+		for i, file := range files {
+			if err := uploadFile(file.Path, uploadUrls[i], file.Name); err != nil {
+				return fmt.Errorf("failed to upload file %s: %w", file.Path, err)
+			}
+		}
+
+		fmt.Println("Successfully uploaded all sourcemaps")
+		return nil
+	}
+}
+
+// verifyApiKey queries the LaunchDarkly Observability API to verify credentials and retrieve the Highlight API key.
+// It takes the LaunchDarkly account ID, project ID, and backend URL as input.
+// Returns:
+//   - string: The Highlight API key used for sourcemap uploads
+//   - string: The LaunchDarkly project ID that was verified
```

Comment:
```suggestion
//   - string: The Highlight project ID that was verified
```
important distinction since this is mapped based on the LaunchDarkly project ID